deploy/ansible/terraform/gcp/terraform.yml

```diff
@@ -5,80 +5,26 @@
     force_init: true
     variables:
       project: "{{project}}"
-      uniqifier: "{{uniqifier}}"
       ssh_user: "{{ssh_user}}"
       ssh_key: "{{ssh_key}}"
+      node_count: "{{node_count|default(3)}}"
   register: terraform_out
 
-- name: Add sf-1 to ansible with a ssh key
+- name: Add sf nodes
   add_host:
-    hostname: sf-1
-    ansible_ssh_host: "{{terraform_out.outputs.sf_1_external.value[0]}}"
-    ansible_ssh_private_key_file: "{{ssh_key_filename}}"
+    hostname: "sf-{{node_idx+1}}"
+    node_name: "sf-{{node_idx+1}}"
+    ansible_ssh_host: "{{item}}"
+    node_ip: "{{item}}"
+    ansible_ssh_private_key_file: "{{ssh_key_filename|default(omit)}}"
     groups: hypervisors, primary, etcd_master
-  when: ssh_key_filename != ""
-
-- name: Add sf-1 to ansible with no ssh key
-  add_host:
-    hostname: sf-1
-    ansible_ssh_host: "{{terraform_out.outputs.sf_1_external.value[0]}}"
-    groups: hypervisors, primary, etcd_master
-  when: ssh_key_filename == ""
-
-- name: Add sf-1 node_ip
-  set_fact:
-    node_name: "sf-1"
-    node_ip: "{{terraform_out.outputs.sf_1_internal.value[0]}}"
-  delegate_to: sf-1
-  delegate_facts: true
-
-- name: Add sf-2 to ansible with a ssh key
-  add_host:
-    hostname: sf-2
-    ansible_ssh_host: "{{terraform_out.outputs.sf_2_external.value[0]}}"
-    ansible_ssh_private_key_file: "{{ssh_key_filename}}"
-    groups: hypervisors, primary, etcd_master
-  when: ssh_key_filename != ""
-
-- name: Add sf-2 to ansible with no ssh key
-  add_host:
-    hostname: sf-2
-    ansible_ssh_host: "{{terraform_out.outputs.sf_2_external.value[0]}}"
-    groups: hypervisors, primary, etcd_master
-  when: ssh_key_filename == ""
-
-- name: Add sf-2 node_ip
-  set_fact:
-    node_name: "sf-2"
-    node_ip: "{{terraform_out.outputs.sf_2_internal.value[0]}}"
-  delegate_to: sf-2
-  delegate_facts: true
-
-- name: Add sf-3 to ansible with a ssh key
-  add_host:
-    hostname: sf-3
-    ansible_ssh_host: "{{terraform_out.outputs.sf_3_external.value[0]}}"
-    ansible_ssh_private_key_file: "{{ssh_key_filename}}"
-    groups: hypervisors, primary, etcd_master
-  when: ssh_key_filename != ""
-
-- name: Add sf-3 to ansible with no ssh key
-  add_host:
-    hostname: sf-3
-    ansible_ssh_host: "{{terraform_out.outputs.sf_3_external.value[0]}}"
-    groups: hypervisors, primary, etcd_master
-  when: ssh_key_filename == ""
-
-- name: Add sf-3 node_ip
-  set_fact:
-    node_name: "sf-3"
-    node_ip: "{{terraform_out.outputs.sf_3_internal.value[0]}}"
-  delegate_to: sf-3
-  delegate_facts: true
+  loop: "{{terraform_out.outputs.sf_nodes_external_ip.value}}"
+  loop_control:
+    index_var: node_idx
```

Comment:
gcp-xl also has the concept of a "primary node", which is where syslog and auth tokens go. It was done because that node is localhost in multinode CI and made log collection easier. I wonder if there's some conditional way we can do that here as well?